# HG changeset patch
# User chegar
# Date 1217350415 25200
#      Tue Jul 29 09:53:35 2008 -0700
# Node ID 629fa51e0281243c7341bce37e4421a7069c72ee
# Parent  31845b0edbd16232f26c266f1f79729fb83a6975
6730740: Fix for 6729881 has apparently broken several 64 bit tests:  "Bad address"
Reviewed-by: alanb, jccollet

diff -r 31845b0edbd1 -r 629fa51e0281 src/solaris/native/java/net/linux_close.c
--- openjdk/jdk/src/solaris/native/java/net/linux_close.c	Wed Jan 21 20:57:52 2015 +0000
+++ openjdk/jdk/src/solaris/native/java/net/linux_close.c	Tue Jul 29 09:53:35 2008 -0700
@@ -281,7 +281,9 @@
 
 int NET_RecvFrom(int s, void *buf, int len, unsigned int flags,
        struct sockaddr *from, int *fromlen) {
-    BLOCKING_IO_RETURN_INT( s, recvfrom(s, buf, len, flags, from, (socklen_t)fromlen) );
+    socklen_t socklen = *fromlen;
+    BLOCKING_IO_RETURN_INT( s, recvfrom(s, buf, len, flags, from, &socklen) );
+    *fromlen = socklen;
 }
 
 int NET_Send(int s, void *msg, int len, unsigned int flags) {
@@ -298,7 +300,9 @@
 }
 
 int NET_Accept(int s, struct sockaddr *addr, int *addrlen) {
-    BLOCKING_IO_RETURN_INT( s, accept(s, addr, (socklen_t)addrlen) );
+    socklen_t socklen = *addrlen;
+    BLOCKING_IO_RETURN_INT( s, accept(s, addr, &socklen) );
+    *addrlen = socklen;
 }
 
 int NET_Connect(int s, struct sockaddr *addr, int addrlen) {
diff -r 31845b0edbd1 -r 629fa51e0281 src/solaris/native/java/net/net_util_md.c
--- openjdk/jdk/src/solaris/native/java/net/net_util_md.c	Wed Jan 21 20:57:52 2015 +0000
+++ openjdk/jdk/src/solaris/native/java/net/net_util_md.c	Tue Jul 29 09:53:35 2008 -0700
@@ -1104,7 +1104,16 @@
     }
 #endif
 
-    rv = getsockopt(fd, level, opt, result, (socklen_t)len);
+#ifdef __solaris__
+    rv = getsockopt(fd, level, opt, result, len);
+#else
+    {
+        socklen_t socklen = *len;
+        rv = getsockopt(fd, level, opt, result, &socklen);
+        *len = socklen;
+    }
+#endif
+
     if (rv < 0) {
         return rv;
     }
